Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -121,24 +124,46 @@ static const AtomString& attachmentSubtitleIdentifier()
     return identifier;
 }
 
+static const AtomString& attachmentSaveIdentifier()
+{
+    static MainThreadNeverDestroyed<const AtomString> identifier("attachment-save"_s);
+    return identifier;
+}
+
+static QualifiedName saveQualifiedName()
+{
+    return QualifiedName { nullAtom(), "save"_s, nullAtom() };
+}
+
+template <typename ElementType>
+RefPtr<ElementType> HTMLAttachmentElement::createShadowElement(const AtomString& id, String textContent)
```

Comment:
Not sure this is the best name, given that it does more than just create the element, it also adds to the container. 

It might be better to just inline the calls to `appendChild`. Especially because right now this method doesn't null check or assert `m_containerElement`. The method is easy to misuse in its current state.

Reply:
I think the factoring is good, we can add an assert if necessary.

Reply:
I'll pass the container by ref, rename the function to `createAndAddElementTo`, and then it won't depend on any member data so it can be made static.

Reply:
I'm not sure I like this new name, the calls don't read like a sentence:
`createAndAddElementTo<HTMLDivElement>(attachmentActionIdentifier(), attachmentActionForDisplay(), *m_containerElement);`

Create and add element to HTML div element, attachment action identifier, action for display.

Reply:
We need ObjC named parameters!
`createElementAndAppendTo`?
`createElementWithIdAndOptionalTextAndAppendTo`? (Getting a bit long...)

Reply:
`createContainedElement(container, id, text)`?